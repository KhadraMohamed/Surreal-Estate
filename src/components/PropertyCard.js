/* eslint-disable prettier/prettier */
import React from "react";
import "../styles/property-card.css";
import PropTypes from "prop-types";

const PropertyCard = (props) => {
  const { title, 
    type, 
    bathrooms, 
    bedrooms, 
    price, 
    city, 
    email 
 } = props;

  return (
    <div className="property-card">
      <div className="title">{title}</div>
      <div className="type">{type}</div>
      <div className="bathrooms">{bathrooms}</div>
      <div className="bedrooms">{bedrooms}</div>
      <div className="price">{price}</div>
      <div className="city">{city}</div>
      <div className="email">{email}</div>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    bedrooms:PropTypes.number.isRequired,
    bathrooms:PropTypes.number.isRequired,
    price:PropTypes.number.isRequired
};
