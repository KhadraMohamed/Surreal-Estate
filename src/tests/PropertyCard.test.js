import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const Props = {
    title: "",
    type: "",
    bathrooms: 0,
    bedrooms: 0,
    price: 0,
    city: "",
    email: "",
  };

  test("renders title correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders type correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders bathrooms correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders bedrooms correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders price correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders city correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders email correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={Props.title}
        type={Props.type}
        bathrooms={Props.bathrooms}
        bedrooms={Props.bedrooms}
        price={Props.price}
        city={Props.city}
        email={Props.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
