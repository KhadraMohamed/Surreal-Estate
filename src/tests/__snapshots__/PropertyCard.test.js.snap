// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders bathrooms correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders bedrooms correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders city correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders email correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders price correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders title correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders type correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
    />
    <div
      class="type"
    />
    <div
      class="bathrooms"
    >
      0
    </div>
    <div
      class="bedrooms"
    >
      0
    </div>
    <div
      class="price"
    >
      0
    </div>
    <div
      class="city"
    />
    <div
      class="email"
    />
  </div>
</DocumentFragment>
`;
